pkg/kubernetes/deployment/extnv1beta1/v1alpha1/rollout_status.go

```diff
@@ -0,0 +1,151 @@
+	// PredicateUpdateInProgress refer to rolloutStatus for predicate IsUpdateInProgress.
+	PredicateUpdateInProgress: func(d *deploy) string {
+		return fmt.Sprintf("Replica update in progress : %d of %d updated replicas are available",
+			d.object.Status.AvailableReplicas, d.object.Status.UpdatedReplicas)
+	},
+	// PredicateNotSpecSynced refer to status rolloutStatus for predicate IsNotSyncSpec.
+	PredicateNotSpecSynced: func(d *deploy) string {
+		return "Deployment rollout in-progress : deployment rollout in-progress : waiting for deployment spec update to be observed"
+	},
+}
+
+// rolloutChecks contains a group of predicate it useses predicateName as key.
+var rolloutChecks = map[predicateName]predicate{
+	// PredicateProgressDeadlineExceeded refer to predicate IsProgressDeadlineExceeded.
+	PredicateProgressDeadlineExceeded: IsProgressDeadlineExceeded(),
+	// PredicateOlderReplicaActive refer to predicate IsOlderReplicaActive.
+	PredicateOlderReplicaActive: IsOlderReplicaActive(),
+	// PredicateTerminationInProgress refer to predicate IsTerminationInProgress.
+	PredicateTerminationInProgress: IsTerminationInProgress(),
+	// PredicateUpdateInProgress refer to predicate IsUpdationInProgress.
+	PredicateUpdateInProgress: IsUpdateInProgress(),
+	// PredicateNotSpecSynced refer to predicate IsSyncSpec.
+	PredicateNotSpecSynced: IsNotSyncSpec(),
+}
+
+// rolloutOutput struct contaons message and boolean value to show rolloutstatus
+type rolloutOutput struct {
+	IsRolledout bool   `json:"isRolledout"`
+	Message     string `json:"message"`
+}
+
+// rawFn is a typed function that abstracts conversion of rolloutOutput struct to raw byte
+type rawFn func(r *rolloutOutput) ([]byte, error)
+
+// rolloutOutputBuilder enables getting various output format of rolloutOutput
+type rolloutOutputBuilder struct {
+	output *rolloutOutput
+	raw    rawFn
+}
+
+// rolloutOutputBuildOption defines the abstraction to build a rolloutOutputBuilder instance
+type rolloutOutputBuildOption func(*rolloutOutputBuilder)
+
+// isRollout range over rolloutChecks map and check status of each predicate
+// also it generates status message from rolloutStatuses using predicate key
+func (d *deploy) isRollout() (string, bool) {
+	msg := ""
+	ok := false
+	for pk, p := range rolloutChecks {
+		if ok = p(d); ok {
+			msg = rolloutStatuses[pk](d)
+			return msg, !ok
+		}
+	}
+	return msg, !ok
+}
+
+// RolloutStatus runs checks against deployment instance
+// and generates rollout status as rolloutOutput
+func (d *deploy) RolloutStatus() (op *rolloutOutput, err error) {
+	op = &rolloutOutput{}
+	msg, ok := d.isRollout()
+	op.IsRolledout = ok
+	if !ok {
+		op.Message = msg
+		return
+	}
+	op.Message = "Deployment successfully rolled out"
+	return
+}
+
+// rolloutStatusf returns a new instance of rolloutOutputBuilder meant for rolloutOutput.
+// caller can configure it with different rolloutOutputBuildOption
+func rolloutStatusf(opts ...rolloutOutputBuildOption) *rolloutOutputBuilder {
```

Comment:
Refer to previous comment on `func NewRollout(opts ...rolloutBuildOption) * rollout`
We have mixed up buildOption function type with Builder struct & pattern.
It need not be mixed.

_NOTE: In other words, builder structure & its methods should not use `xyzBuildOption`._

Reply:
done please check.